autofs-5.1.7 - fix dandling symlink creation if nis support is not available

From: Ian Kent <raven@themaw.net>

If NIS support is not available a dangling symlink is created pointing
from lookup_nis.so to (a non-existent) lookup_yp.so.

Signed-off-by: Ian Kent <raven@themaw.net>
---
 CHANGELOG        |    1 +
 modules/Makefile |    2 ++
 2 files changed, 3 insertions(+)

diff --git a/CHANGELOG b/CHANGELOG
index ddc07912..06bf24b8 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -67,6 +67,7 @@
 - cater for empty mounts list in mnts_get_expire_list().
 - add ext_mount_hash_mutex lock helpers.
 - fix amd section mounts map reload.
+- fix dandling symlink creation if nis support is not available.
 
 25/01/2021 autofs-5.1.7
 - make bind mounts propagation slave by default.
diff --git a/modules/Makefile b/modules/Makefile
index 91dc20ab..6908da06 100644
--- a/modules/Makefile
+++ b/modules/Makefile
@@ -77,7 +77,9 @@ install: all
 	install -c $(MODS) -m 755 $(INSTALLROOT)$(autofslibdir)
 	-rm -f $(INSTALLROOT)$(autofslibdir)/mount_smbfs.so
 	ln -fs lookup_file.so $(INSTALLROOT)$(autofslibdir)/lookup_files.so
+ifeq ($(YPCLNT), 1)
 	ln -fs lookup_yp.so $(INSTALLROOT)$(autofslibdir)/lookup_nis.so
+endif
 ifeq ($(LDAP), 1)
 	ln -fs lookup_ldap.so $(INSTALLROOT)$(autofslibdir)/lookup_ldaps.so
 endif
